go-controller/hybrid-overlay/pkg/controller/node_linux.go

```diff
@@ -65,11 +65,14 @@ type NodeController struct {
 //  1. Setting up a VXLAN gateway and hooking to the OVN gateway
 //  2. Setting back annotations about its VTEP and gateway MAC address to its own object
 func newNodeController(
-	_ kube.Interface,
+	kube kube.Interface,
 	nodeName string,
 	nodeLister listers.NodeLister,
 	localPodLister listers.PodLister,
 ) (nodeController, error) {
+	if kube != nil {
+		return newHONodeController(kube, nodeName, nodeLister, localPodLister)
+	}
```

Comment:
I am confused here, if we are completely replacing what the hybrid overlay binary does on linux, why don't we just come up with a new binary? Wouldn't this prevent us to use the HO binary on linux for the old testing puposes that we had in the past if we ever wanted to again?

Reply:
What do you mean when you say `I want the hybrid-overlay-node to do the same work on both Windows and Linux nodes`? I guess if they were, they could both re-use the same code but they don't.

It's not important other than the weirdness. 

But where I think we could improve is making the difference between both linux controllers more intuitive, either through better naming, better godoc text or both:

I can hardly understand the difference, without deep diving into the code, between the one that was already there
```
// NodeController is the node hybrid overlay controller
type NodeController struct {
``` 
and the new one
```
// HONodeController is the node hybrid overlay controller
type HONodeController struct {
```

Reply:
The hybrid-overlay-node Linux binary was not used before. So I guess nobody cares what it does. Now we want to run hybrid-overlay-node Linux binary at nodes that are not managed by OVN-K.

I've updated the comment in the code. PTAL.